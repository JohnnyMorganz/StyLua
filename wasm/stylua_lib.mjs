import fs from "node:fs";
import { initSync } from "./stylua.web/stylua_lib.js";

const wasm = new URL("./stylua.web/stylua_lib_bg.wasm", import.meta.url);
initSync(fs.readFileSync(wasm));

export * from "./stylua.web/stylua_lib.js";
