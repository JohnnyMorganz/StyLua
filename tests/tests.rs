use stylua_lib::{format_code, CollapseSimpleStatement, Config, OutputVerification};

fn format(input: &str) -> String {
    format_code(input, Config::default(), None, OutputVerification::None).unwrap()
}

#[test]
#[cfg_attr(feature = "luau", ignore)]
fn test_standard() {
    insta::glob!("inputs/*.lua", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(format(&contents));
    })
}

#[test]
#[cfg_attr(feature = "lua52", ignore)] // A test case has `goto` as an identifier, which is not allowed in Lua 5.2
fn test_full_moon_test_suite() {
    insta::glob!("inputs-full_moon/*.lua", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(format(&contents));
    })
}

#[test]
#[cfg(feature = "luau")]
fn test_luau() {
    insta::glob!("inputs-luau/*.lua", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(format(&contents));
    })
}

#[test]
#[cfg(feature = "luau")]
fn test_luau_full_moon() {
    insta::glob!("inputs-luau-full_moon/*.lua", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(format(&contents));
    })
}

#[test]
#[cfg(feature = "lua52")]
fn test_lua52() {
    insta::glob!("inputs-lua52/*.lua", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(format(&contents));
    })
}

#[test]
fn test_ignores() {
    insta::glob!("inputs-ignore/*.lua", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(format(&contents));
    })
}

#[test]
fn test_collapse_single_statement() {
    insta::glob!("inputs-collapse-single-statement/*.lua", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(format_code(
            &contents,
            Config::default().with_collapse_simple_statement(CollapseSimpleStatement::Always),
            None,
            OutputVerification::None
        )
        .unwrap());
    })
}
