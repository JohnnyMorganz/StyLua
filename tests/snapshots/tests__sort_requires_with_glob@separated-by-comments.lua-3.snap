---
source: tests/tests.rs
expression: "format!(\"glob_pattern: {}\\n{}\", glob_pattern,\nformat_code(&contents, Config\n{\n    sort_requires: SortRequiresConfig\n    { enabled: true, glob: Some(glob_pattern.to_string()) },\n    ..Config::default()\n}, None, OutputVerification::None).unwrap())"
input_file: tests/inputs-sort-requires/separated-by-comments.lua
---
glob_pattern: S*
-- Services
local A = require("A")
local B = require("B")
local C = require("C")

-- Packages
local X = require("X")
local Y = require("Y")
local Z = require("Z")
